core/src/main/java/org/apache/iceberg/jdbc/JdbcTableOperations.java

```diff
@@ -251,4 +255,14 @@ private Map<String, String> getTable()
           return table;
         });
   }
+
+  /**
+   * Check that pool is off and turn it back on if it is off, in case others turn it off and cannot
+   * use pool.
+   */
+  private void checkIsClosedOtherwiseReuse() {
+    if (connections != null && connections.isClosed()) {
```

Comment:
I think we can confirm whether it is closed before each use. If it is closed, we need to open it again and use it again